R/chart_hc_imd_bar.R

```diff
@@ -0,0 +1,220 @@
+#' \itemize{
+#'  \item Default = ""
+#'  \item k:- values will be divided by 1,000 and figures will include "k" suffix
+#'  \item m:- values will be divided by 1,000,000 and figures will include "m" suffix
+#'  \item bn:- values will be divided by 1,000,000,000 and figures will include "bn" suffix
+#'  \item tn:- values will be divided by 1,000,000,000,000 and figures will include "tn" suffix
+#' }
+#'
+#' @return Highchart object for bar chart showing data by IMD decile
+#'
+#' @example man/examples/chart_hc_imd_bar.R
+#'
+#' @export
+chart_hc_imd_bar <- function(
+    df,
+    imd_col,
+    value_col,
+    metric_def,
+    highlight_core20 = FALSE,
+    show_lbl = FALSE,
+    value_dp = 0,
+    value_prefix = "",
+    order_of_magnitude = ""
+) {
+
+# Input review --------------------------------------------------------------------------------
+
+  # review input parameters, providing error/warning messages as required and aborting call
+
+  # highlight_core20
+  if(!highlight_core20 %in% c(TRUE,FALSE)){
+    stop("Invalid parameter value (highlight_core20) : Must be one of TRUE/FALSE", call. = FALSE)
+  }
+
+  # show_lbl
+  if(!show_lbl %in% c(TRUE,FALSE)){
+    stop("Invalid parameter value (show_lbl) : Must be one of TRUE/FALSE", call. = FALSE)
+  }
+
+  # value_dp
+  if(!value_dp %in% c(0,1,2,3)){
+    stop("Invalid parameter value (value_dp) : Must be one of 0/1/2/3", call. = FALSE)
+  }
+
+  # order_of_magnitude
+  if(!order_of_magnitude %in% c("","k","m","bn","tn")){
+    stop("Invalid parameter value (order_of_magnitude) : Must be blank or one of k/m/bn/tn", call. = FALSE)
+  }
+
+  # IMD column entries (10 rows with values 1 to 10)
+  # if(nrow(df) != 10){
+  if(nrow(df) != 10 || !all(sort(as.integer(df[[imd_col]])),seq(1:10))){
+    stop("Invalid IMD decile values : Data must contain 10 rows with IMD decile values from 1 to 10", call. = FALSE)
+  }
+
+
+# Data object ---------------------------------------------------------------------------------
+
+  # define the data object based on parameters
+  df <- df |>
+    dplyr::mutate(IMD_COL = df[[imd_col]],
+                  VAL_COL = df[[value_col]])
+
+  # define grouping based on parameters
+  if(highlight_core20 == TRUE){
+    df <- df |>
+      dplyr::mutate(GRP_COL = dplyr::case_when(IMD_COL %in% c(1,2) ~ "Core20",
+                                               TRUE ~ "Non-Core20")
+                    )
+  } else {
+    df <- df |>
+      dplyr::mutate(GRP_COL = metric_def)
+  }
+
+  # adjust values based on supplied order of magnitude
+  if(order_of_magnitude != ""){
+    df <- df |>
+      dplyr::mutate(VAL_COL = VAL_COL / switch(order_of_magnitude,
+                                               "k" = 1000,
+                                               "m" = 1000000,
+                                               "bn" = 1000000000,
+                                               "tn" = 1000000000000)
+                    )
+  }
```

Comment:
Alternative way:

change arg to `order_of_magnitude = "identity"`

```
magnitudes <- c(identity = 10^0, k = 10^3, m = 10^6, bn = 10^9, tn = 10^12)

df <- df |>
  dplyr::mutate(VAL_COL = VAL_COL / magnitudes[[order_of_magnitude]])
```

Reply:
I see you use the fact that order_of_magnitude is `""` by default later, so would have to add below after. That is a bit icky, so maybe not use my suggestion!

`order_of_magnitude <- ifelse(order_of_magnitude = "identity", "", order_of_magnitude)`